lib/src/observable.dart

```diff
@@ -1,92 +1,131 @@
+/// https://dart.dev/articles/libraries/code/stream_controller.dart
+extension ObservableStore on Store {
+  /// Create a stream to data changes on EntityT (stored Entity class).
+  ///
+  /// The stream receives an event whenever an object of EntityT is created or
+  /// changed or deleted. Make sure to close() the subscription after you're
+  /// done with it to avoid hanging change listeners.
+  Stream<void> subscribe<EntityT>() {
+    final observer = _Observer<void>();
+    final entityId = entityDef<EntityT>().model.id.id;
+
+    observer.init(() {
+      // We're listening to events on single entity so there's no argument.
+      // Ideally, controller.add() would work but it doesn't, even though we're
+      // using StreamController<Void> so the argument type is `void`.
+      observer.receivePort = ReceivePort()
+        ..listen((_) => observer.controller.add(null));
+      observer.cObserver = bindings.obx_dart_observe_single_type(
+          ptr, entityId, observer.nativePort);
+    });
+
+    return observer.stream;
   }
 
+  /// Create a stream to data changes on all Entity types.
+  ///
+  /// The stream receives an even whenever any data changes in the database.
+  /// Make sure to close() the subscription after you're done with it to avoid
+  /// hanging change listeners.
+  Stream<Type> subscribeAll() {
+    initializeDartAPI();
+    final observer = _Observer<Type>();
+
+    // create a map from Entity ID to Entity type (dart class)
+    final entityTypesById = <int, Type>{};
+    defs.bindings.forEach((Type entity, EntityDefinition entityDef) =>
+        entityTypesById[entityDef.model.id.id] = entity);
+
+    observer.init(() {
+      // We're listening to a events for all entity types. C-API sends entity ID
+      // and we must map it to a dart type (class) corresponding to that entity.
+      observer.receivePort = ReceivePort()
+        ..listen((entityIds) {
+          if (entityIds is! List) {
+            observer.controller.addError(Exception(
+                'Received invalid data format from the core notification: (${entityIds.runtimeType}) $entityIds'));
+            return;
+          }
+
+          entityIds.forEach((entityId) {
+            if (entityId is! int) {
+              observer.controller.addError(Exception(
+                  'Received invalid item data format from the core notification: (${entityId.runtimeType}) $entityId'));
+              return;
+            }
+
+            final entityType = entityTypesById[entityId];
+            if (entityType == null) {
+              observer.controller.addError(Exception(
+                  'Received data change notification for an unknown entity ID $entityId'));
+            } else {
+              observer.controller.add(entityType);
+            }
+          });
+        });
+      observer.cObserver = bindings.obx_dart_observe(ptr, observer.nativePort);
+    });
+
+    return observer.stream;
+  }
+}
+
+extension Streamable<T> on Query<T> {
```

Comment:
Could you add method `countStream`?

Reply:
You could also count on the query from the queryStream, right?

Reply:
For example:
```
Stream<int> get countStream => store.subscribe<T>().map((_) => count());
```
@Buggaboo As I understand `count()`, this is a lighter operation than pull all the data with `find`

Reply:
It would also be great to have:
```
Stream<T> get findFirstStream => store.subscribe<T>().map((_) => findFirst());
```

Reply:
This is just a suggestion for improvement.

Reply:
Of course. I appreciate it. You could branch off from @vaind, and offer a PR to his branch, I suspect he's too busy atm.

Reply:
@Buggaboo So I understand. And if @vaind doesn't mind I can create PR for this branch.

Reply:
would wait with that, I'm still changing & more importantly rebasing this so it would mess up your fork

Reply:
Recently, I've been looking at this a little differently, why create a stream-specific method for everything when you can do the same easily with query.stream.map(), I've created a follow up issue #195

Reply:
> Recently, I've been looking at this a little differently, why create a stream-specific method for everything when you can do the same easily with query.stream.map(), I've created a follow up issue #195

But query stream always fetch data...